inc/parser/handler.php

```diff
@@ -363,22 +363,92 @@ function quote($match, $state, $pos) {
         return true;
     }
 
+    /**
+     * Internal function for parsing highlight options.
+     * $options is parsed for key value pairs separated by commas.
+     * A value might also be missing in which case the value will simple
+     * be set to true. Commas in strings are ignored, e.g. option="4,56"
+     * will work as expected and will only create one entry.
+     *
+     * @param string $options Comma separated list of key-value pairs,
+     *                        e.g. option1=123, option2="456"
+     * @return array|null     Array of key-value pairs $array['key'] = 'value';
+     *                        or null if no entries found
+     */
+    protected function parse_highlight_options ($options) {
+        $result = array();
+        preg_match_all('/(\w+(?:="[^"]*"))|(\w+[^=,\]])(?:,*)/', $options, $matches, PREG_SET_ORDER);
+        foreach ($matches as $match) {
```

Comment:
Is this parser needed? At first glance the syntax looks like json. What about using the json parser?

Reply:
What's the consensus here? Do we keep the parser or do we swap to JSON (means a change is requested)?

Reply:
I think it makes sense to replace this by `json_decode()` it makes the code less complex and easier to understand.

Reply:
Done.